packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -680,8 +681,12 @@ export function getPrismaClient(config: GetPrismaClientConfig) {
      * @returns
      */
     $executeRaw(query: TemplateStringsArray | sqlTemplateTag.Sql, ...values: any[]) {
-      return createPrismaPromise((txId, lock, otelCtx) => {
-        if ((query as TemplateStringsArray).raw || (query as sqlTemplateTag.Sql).sql) {
+      return createPrismaPromise(async (txId, lock, otelCtx): Promise<unknown> => {
+        if (isSql(query) && query.sql === '') {
+          return null
+        }
+
+        if ((query as TemplateStringsArray).raw !== undefined || (query as sqlTemplateTag.Sql).sql !== undefined) {
```

Comment:
I wonder if the previous (the newly added) check is now necessary or just adding `!== undefined` here is enough to fix the bug.

Reply:
I think the check can now be removed

Reply:
This check needs to be here because we cannot send an empty string to the DB. So the newly added check returns null if the string is empty.

```
Invalid `prisma.$executeRaw()` invocation:


      Raw query failed. Code: `1065`. Message: `Query was empty`

      48980 |       message = this.sanitizeMessage(message);
      48981 |       if (e.code) {
    > 48982 |         throw new PrismaClientKnownRequestError(message, e.code, this.client._clientVersion, e.meta);
            |               ^
      48983 |       } else if (e.isPanic) {
      48984 |         throw new PrismaClientRustPanicError(message, this.client._clientVersion);
      48985 |       } else if (e instanceof PrismaClientUnknownRequestError) {

      at RequestHandler.request (../../runtime/index.js:48982:15)
      at Proxy._request (../../runtime/index.js:49888:18)
      at Object.<anonymous> (empty-to-raw/tests.ts:11:22)
```

Reply:
Sure it could be seen as breaking! I have reverted the changes I made and instead now assert the error could be thrown or the `0` or the `[]`. My reason for adding the null return is because it's deterministic, easier to reason about/document plus the issue expected 'to run it without any errors. preferably don't issue any queries if the input is Prisma.empty'.